plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLCatalog.java

```diff
@@ -626,10 +627,27 @@ protected MySQLTableBase fetchObject(@NotNull JDBCSession session, @NotNull MySQ
             if (tableType != null && tableType.contains("VIEW")) {
                 return new MySQLView(owner, dbResult);
             } else {
-                return new MySQLTable(owner, dbResult);
+                MySQLTable mySQLTable = new MySQLTable(owner, dbResult);
+                loadAdditionalInfoIfNeed(owner, mySQLTable);
+                return mySQLTable;
             }
         }
 
+        private void loadAdditionalInfoIfNeed(@NotNull MySQLCatalog owner, @NotNull MySQLTable mySQLTable) {
```

Comment:
This is bad... So it refreshes the additional info of the old instance of `MySQLTable` that `mySQLTable` is supposed to replace? Shouldn't the new object replace the one in the cache? How does it work even?

Reply:
It also breaks the idea of the additional info being lazy to load.

Reply:
I don't get it. Won't the navigator detect that the object doesn't have its statistics cached, because it's a new object, and load it?